File: n-back_bastien/SequenceOfLetterTB.m
function [finalindexletter]=SequenceOfLetterTB(lseq)

if mod(lseq,2)~=0
    error('length of sequence has to be divisible per 2');
else
    indexlettertmp=[ones(1,lseq/2) ones(1,lseq/2)+1];
    finalindexletter=indexlettertmp(randperm(lseq));
end

File: n-back_bastien/SequenceOfLetterTB.py
import random

def SequenceOfLetterTB(lseq):
    if lseq % 2 != 0:
        raise ValueError("Length of sequence has to be divisible by 2")
    else:
        index_letter_tmp = [1] * (lseq // 2) + [2] * (lseq // 2)
        random.shuffle(index_letter_tmp)
        return index_letter_tmp

# test = SequenceOfLetterTB(10)
# print(test)
